packages/node_modules/@webex/plugin-meetings/src/media/detectSilence/videoSilence.js

```diff
@@ -0,0 +1,48 @@
+/*
+  Code influenced from:
+    - https://stackoverflow.com/questions/46543341/how-can-i-extract-the-preceding-audio-from-microphone-as-a-buffer-when-silence
+    - https://github.com/twilio/twilio-video.js/blob/master/lib/util/detectsilentvideo.js
+    - https://github.com/twilio/twilio-video.js/blob/master/lib/util/detectsilentaudio.js
+    - https://github.com/twilio/twilio-video.js/blob/master/lib/webaudio/detectsilence.js
+*/
+
+import {
+  DETECT_VIDEO_SILENCE_SAMPLE_HEIGHT,
+  DETECT_VIDEO_SILENCE_SAMPLE_WIDTH
+} from '../../constants';
+
+/**
+ * Check whether the current video frame is silent by selecting a 50x50
+ * sample and calculating the max value of the pixel data. If it is 0, then
+ * the frame is considered to be silent.
+ *
+ * @private
+ * @param {HTMLMediaElement | HTMLVideoElement} el
+ * @param {canvas} canvas
+ * @returns {boolean} true if silent, false if not
+ */
+const checkVideoSilence = (el, canvas) => {
+  const context = canvas.getContext('2d');
+
+  context.drawImage(
+    el,
+    0,
+    0,
+    DETECT_VIDEO_SILENCE_SAMPLE_WIDTH,
+    DETECT_VIDEO_SILENCE_SAMPLE_HEIGHT
+  );
+
+  const frame = context.getImageData(
+    0,
+    0,
+    DETECT_VIDEO_SILENCE_SAMPLE_WIDTH,
+    DETECT_VIDEO_SILENCE_SAMPLE_HEIGHT
+  );
+
+  const frameDataWithoutAlpha = frame.data.filter((item, i) => (i + 1) % 4);
+  const max = Math.max(...frameDataWithoutAlpha);
+
+  return max === 0;
```

Comment:
this could be optimised quite a lot. Currently we're doing 2 passes over the whole array of data, while we could do it in 1 pass and stop on the first non-zero value (ignoring alpha)